### PR TITLE
ImmutableSet::putAll, of for instantiation

### PR DESCRIPTION
**Pull request by Protryon**
Prevents verbose manual creation of sets with default entries.

### CODE REVIEW

File: src/main/java/com/shapesecurity/functional/data/ImmutableSet.java [hunk 1]
```diff
@@ -53,6 +71,15 @@ public int length() {
         return list.foldLeft(ImmutableSet::put, this);
     }
 
+    @Nonnull
+    public <B extends T> ImmutableSet<T> putAll(@Nonnull B... list) {
```
Comment (bakkot) [inline src/main/java/com/shapesecurity/functional/data/ImmutableSet.java]: Not sure we need this. Calling `put` multiple times is not that much more verbose, and is not ambiguous with the other `putAll`.

Comment (Protryon) [reply src/main/java/com/shapesecurity/functional/data/ImmutableSet.java]: This is useful for array types (they can take the place of rest params in Java), and results in cleaner default/static lists. I don't believe the ambiguity will be an issue here, but we can call it something else.

Comment (bakkot) [reply src/main/java/com/shapesecurity/functional/data/ImmutableSet.java]: I dislike the overload. If you can come up with a better name I'm OK with it.

**Comment by michaelficarra** [comment]
Actual `putAll` could be made to take an `Iterable`.

**Pull request merged**